worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,68 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e) {
+    if (!e.application.getInputMessageProcessor().getProcessSubdocumentsSeparately() && e.rootDocument.hasSubdocuments()) {
+        processSubdocumentFailures(e.rootDocument.getSubdocuments());
+    }
+    processFailures(e.document);
+}
+
+function processSubdocumentFailures(subdocuments) {
+    for each(var subdoc in subdocuments) {
+        if (subdoc.hasSubdocuments()) {
+            processSubdocumentFailures(subdoc.getSubdocuments());
+        } else {
+            processFailures(subdoc);
+        }
+    }
+}
+
+function processFailures(document) {
+    if (document.getFailures().isChanged()) {
+
+        var listOfFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfFailures.add(failure);
+        });
+
+        document.getFailures().reset();
+
+        var listOfOriginalFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfOriginalFailures.add(failure);
+        });
+
+        for each (var f in listOfFailures) {
+            if (!isFailureInOriginal(listOfOriginalFailures, f)) {
+                var source = document.rootDocument.getField("CAF_WORKFLOW_ACTION").getStringValues().get(0);
+                var numericVersion = document.getTask().getService(com.hpe.caf.api.worker.WorkerTaskData.class).getSourceInfo().getVersion();
+                var message = {
+                    ID: f.getFailureId(),
+                    STACK: f.getFailureStack() || undefined,
+                    WORKFLOW_ACTION: source,
+                    VERSION: source.trim() + numericVersion.trim(),
```

Comment:
- Source and action are getting confused here - for this use the actual source rather than the workflow action
- Put a space between the source and version